base/bucket_gocb.go

```diff
@@ -1386,6 +1586,146 @@ func (bucket CouchbaseBucketGoCB) CouchbaseServerVersion() (major uint64, minor
 }
 
 func (bucket CouchbaseBucketGoCB) UUID() (string, error) {
+
+	// Temp workaround -- create a go-couchbase bucket just to get the UUID
 	// See https://github.com/couchbase/sync_gateway/issues/2418#issuecomment-289941131
-	return "error", fmt.Errorf("GoCB bucket does not expose UUID")
+	goCouchbaseBucket, err := GetCouchbaseBucket(bucket.spec, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return goCouchbaseBucket.UUID()
+
+}
+
+func (bucket CouchbaseBucketGoCB) Close() {
+	if err := bucket.Bucket.Close(); err != nil {
+		Warn("Error closing GoCB bucket: %v", err)
+	}
+
+}
+
+// Applies the viewquery options as specified in the params map to the viewQuery object,
+// for example stale=false, etc.
+func applyViewQueryOptions(viewQuery *gocb.ViewQuery, params map[string]interface{}) {
+
+	for optionName, optionValue := range params {
+		switch optionName {
+		case ViewQueryParamStale:
+			optionAsBool := asBool(optionValue)
+			if optionAsBool == false {
+				viewQuery.Stale(gocb.Before)
+			}
+		case ViewQueryParamReduce:
+			viewQuery.Reduce(asBool(optionValue))
+		case ViewQueryParamLimit:
+			uintVal, err := normalizeIntToUint(optionValue)
+			if err != nil {
+				Warn(fmt.Sprintf("%v", err))
+			}
+			viewQuery.Limit(uintVal)
+		case ViewQueryParamDescending:
+			if asBool(optionValue) == true {
+				viewQuery.Order(gocb.Descending)
+			}
+		case ViewQueryParamSkip:
+			uintVal, err := normalizeIntToUint(optionValue)
+			if err != nil {
+				Warn(fmt.Sprintf("%v", err))
+			}
+			viewQuery.Skip(uintVal)
+		case ViewQueryParamGroup:
+			viewQuery.Group(asBool(optionValue))
+		case ViewQueryParamGroupLevel:
+			uintVal, err := normalizeIntToUint(optionValue)
+			if err != nil {
+				Warn(fmt.Sprintf("%v", err))
+			}
+			viewQuery.GroupLevel(uintVal)
+		case ViewQueryParamKey:
+			viewQuery.Key(optionValue)
+		case ViewQueryParamKeys:
+			stringKeys := optionValue.([]string)
+			emptyInterfaceKeys := []interface{}{}
+			for _, key := range stringKeys {
+				emptyInterfaceKeys = append(emptyInterfaceKeys, key)
+			}
+			viewQuery.Keys(emptyInterfaceKeys)
+		}
```

Comment:
Fixed in 29042b197214d4c2f458fc0097e80113b3df9e8f -- it now returns an error, and I added case branches to explicitly cover items that are ignored or dealt with outside of the switch statement.